runtime/common/src/main/kotlin/kotlinx/serialization/context/SerialModules.kt

```diff
@@ -0,0 +1,168 @@
+@file:Suppress("RedundantVisibilityModifier")
+
+package kotlinx.serialization.context
+
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import kotlin.reflect.KClass
+
+/**
+ * SerialModule is a collection of classes associated with its serializers,
+ * postponed for runtime resolution. Its single purpose is to register all
+ * serializers it has in the given [MutableSerialContext].
+ *
+ * Typically, one can create a module (using library implementation or anonymous class)
+ * per file, or per package, to hold
+ * all serializers together and then register it in some [AbstractSerialFormat].
+ *
+ * @see AbstractSerialFormat.install
+ */
+interface SerialModule {
+
+    /**
+     * Registers everything it has in the [context].
+     *
+     * @see MutableSerialContext.registerSerializer
+     * @see MutableSerialContext.registerPolymorphicSerializer
+     */
+    fun registerIn(context: MutableSerialContext)
+}
+
+/**
+ * A [SerialModule] which registers one class with one serializer for [ContextSerializer].
+ *
+ * @see MutableSerialContext.registerSerializer
+ */
+class SimpleModule<T: Any>(val kClass: KClass<T>, val kSerializer: KSerializer<T>): SerialModule {
```

Comment:
`Simple` is a polluted word. Consider e.g. `SingletonModule`.

Its usage looks a bit boilerplate:
```
Json().apply { install( SimpleModule(Payload::class, PayloadSerializer)) }
```

what if we introduce extensions on serial format such as
```
Json().apply { install(Payload::class, PayloadSerializer) } // 1
Json().register(Payload::class, PayloadSerializer) // 2
```

If we could provide a reified version, even better: `Json().register(PayloadSerializer)`

Reply:
>WDYT about passing SerialModule as a constructor parameter in all serial formats?

 I am fine with it